libtiny_client/src/lib.rs

```diff
@@ -660,3 +695,127 @@ async fn try_connect(
 
     None
 }
+
+/// Spawns a task to connect to given address to download
+/// a file
+/// https://www.irchelp.org/protocol/ctcpspec.html
+fn dcc_file_get(
+    msg_chan: &mut mpsc::Sender<Cmd>,
+    addr: SocketAddr,
+    mut download_dir: PathBuf,
+    file_name: String,
+    target: String,
+) {
+    let mut msg_chan = msg_chan.clone();
+    tokio::task::spawn_local(async move {
+        let stream = match timeout(
+            tokio::time::Duration::from_secs(25),
+            tokio::net::TcpStream::connect(addr),
+        )
+        .await
+        {
+            Err(err) => {
+                error!("{}", err);
+                let err_msg = "File transfer connection timed out.";
```

Comment:
A strange issue is happening here where the error message gets truncated to "File". I was wondering if it has to do with the mpsc buffer, so I changed `try_send` to `send` but the same thing happened. It **does** send the correct message when the file transfer is successful.